child and task state passing from api call

Schrese opened the pull request:
# Description
Passes state for children as well as their tasks throughout the app. 
Also updated the UploadDocs component to disable the submit button, limit the number of uploads, and added a notification/popup for having too many images

Fixes #

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)


## Change Status

- [x] Complete, tested, ready to review and merge


## Has This Been Tested

- [x] Yes


## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My code has been reviewed by at least one peer
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] There are no merge conflicts

Schrese committed:
child and task state passing from api call

Schrese committed:
added some documentation to api call

Schrese committed:
committing before pulling from main

Schrese committed:
Merge branch 'main' of https://github.com/Lambda-School-Labs/Labs26-StorySquad-FE-TeamB into feature/child-state-management

Schrese committed:
state passes from child dash to mission-control

Schrese committed:
created api calls for marking read task complete

Schrese committed:
Merge branch 'main' of https://github.com/Lambda-School-Labs/Labs26-StorySquad-FE-TeamB into feature/child-state-management

Schrese committed:
reading finished button now works

SandraCoburn commented on src/state/reducers/taskReducer.js:
```diff
@@ -13,12 +13,12 @@ export const reducer = (state = initialState, action) => {
+        story_id: action.payload.StoryId,
+        hasRead: action.payload.HasRead,
+        hasWritten: action.payload.HasWritten,
+        hasDrawn: action.payload.HasDrawn,
```
wouldn't you need to make a copy of state before modifying? 
return {...state, and then what you want to modify}

bramirez96 replied:
In this case it should be fine because every field is being set in the return statement!

Schrese committed:
loading checkboxes is faster

Schrese committed:
getting rid of button for pull from main

Schrese committed:
Merge branch 'main' of https://github.com/Lambda-School-Labs/Labs26-StorySquad-FE-TeamB into feature/child-state-management

Schrese committed:
Updated styling of mission control and read button

Schrese committed:
writing submission functional

bramirez96 committed:
Updated way that button disabled state is managed on story viewer.

Schrese committed:
api is now only called once

Schrese committed:
Merge branch 'feature/child-state-management' of https://github.com/Lambda-School-Labs/Labs26-StorySquad-FE-TeamB into feature/child-state-management

Schrese committed:
hasRead state now works

Schrese committed:
Writing submission now works and holds state

Schrese committed:
Merge branch 'main' of https://github.com/Lambda-School-Labs/Labs26-StorySquad-FE-TeamB into feature/child-state-management

Schrese committed:
Drawing submission and state works

Schrese committed:
removed console.logs, fixed multiple upload

Schrese committed:
updated jsdocs for api calls
dissabled upload button for multiple images on drawing submission

Schrese committed:
correct tests, adds warning notification to upload

Schrese committed:
Merge branch 'main' of https://github.com/Lambda-School-Labs/Labs26-StorySquad-FE-TeamB into feature/child-state-management

bramirez96 commented on src/components/common/UploadDocs.js:
```diff
@@ -30,6 +37,10 @@ const UploadDocs = ({
     title: '',
   });
 
+  const [maxLength, setMaxLength] = useState(1);
```
We can pass this in as a prop instead of using a state hook!

bramirez96 commented on src/components/common/UploadDocs.js:
```diff
@@ -18,6 +21,10 @@ const UploadDocs = ({
   submissionId,
+  storyId,
+  setHasWritten,
+  setHasDrawn,
```
We can pass a reference to a function instead of passing both functions in and conditionally calling them!

bramirez96 reviewed (changes requested):
Let's keep the UploadDocs component modular! I've marked a few potential changes to help with this

SandraCoburn reviewed (dismissed):
That's a lot of code! Good job.

Schrese committed:
fixed read button
fixed remove on upload
made changes to fall in line with reusability of uploadDocs component

Schrese committed:
added field to tasks, fixed class in pdf viewer

Schrese committed:
changed UploadDocs to not be connected to redux

Schrese committed:
fixed tests

SandraCoburn reviewed (approved):
Thanks for fixing the tests!

bramirez96 reviewed (approved):
Nice work on all of this! I know it was a lot you CRUSH'D IT
